src/character.cpp

```diff
@@ -1289,6 +1289,23 @@ std::array<encumbrance_data, num_bp> Character::calc_encumbrance( const item &ne
     return ret;
 }
 
+int Character::get_weight() const
+{
+    int ret = 0;
+    int wornWeight;
+    wornWeight = std::accumulate(worn.begin(),
+                                 worn.end(),
+                                 0,
+                                 [](int sum, const item& itm) { return sum + itm.weight(); }
+                                 );
```

Comment:
Do you mean like this?

```C++
    wornWeight = std::accumulate(worn.begin(), worn.end(), 0,
                                 [](int sum, const item& itm)
                                 { return sum + itm.weight(); }
                                 );
```

Also, I've just realized there is a Character::weight_carried function that does basically the same thing as I'm trying to. Creature::get_weight + Character::weight_carried = Character::get_weight. It doesn't use the fancy accumulate method though.